code/datums/tutorial/marine/hospital_corpsman_sandbox.dm

```diff
@@ -0,0 +1,682 @@
+	/// Wave number when the last resupply phase triggered. Will wait 3 waves before rolling again
+	var/last_resupply_round = 1
+	/// Wave number when the last mass-casualty (nightmare) phase triggered. Will wait 3 waves before rolling again
+	var/last_masscas_round = 1
+	/// Lists possible chemicals that can appear pre-medicated in patient NPCs in harder difficulties
+	var/list/datum/reagent/medical/premeds = list(/datum/reagent/medical/tramadol, /datum/reagent/medical/bicaridine, /datum/reagent/medical/kelotane, /datum/reagent/medical/oxycodone)
+
+/datum/tutorial/marine/hospital_corpsman_sandbox/start_tutorial(mob/starting_mob)
+	. = ..()
+	if(!.)
+		return
+
+	START_PROCESSING(SSfastobj, src)
+	init_mob()
+	init_npcs()
+	message_to_player("Welcome to the Hospital Corpsman tutorial sandbox mode!")
+	message_to_player("Gear up in your prefered HM kit, then press the orange 'Ready Up' arrow at the top of your HUD to begin the first round!")
+
+/datum/tutorial/marine/hospital_corpsman_sandbox/proc/handle_round_progression()
+
+	var/difficulty_upgrade_warning = null
+
+	if(booboo_timer)
+		deltimer(booboo_timer)
+		booboo_timer = null
+
+	switch(stage)
+		if(TUTORIAL_HM_PHASE_RESUPPLY)
+			return // sometimes it double-calls handle_round_progression()
+		// Ensures that a resupply always follows a mass-cas, disregarding minimum rounds between
+		// Lowers difficulty close to, if not slightly below baseline
+		if(TUTORIAL_HM_PHASE_NIGHTMARE)
+			stage = TUTORIAL_HM_PHASE_RESUPPLY
+			survival_wave++
+			survival_difficulty = pick(TUTORIAL_HM_INJURY_SEVERITY_MINOR, TUTORIAL_HM_INJURY_SEVERITY_ROUTINE)
+			min_survival_agents = 1
+			max_survival_agents = 3
+			begin_supply_phase()
+			return
+	// 1 in 5 chance per round to trigger a resupply phase
+	// Will only roll beyond wave 3, and 3 waves after the previous resupply phase.
+	if((rand() < (1/5)) && (survival_wave >= (last_resupply_round + 3)))
+		begin_supply_phase()
+		last_resupply_round = survival_wave
+		return
+	survival_wave++
+	// 1 in 10 chance per round to trigger a mass-cas (NIGHTMARE) phase
+	// Will only roll beyond wave 3, and 3 waves after the previous mass-cas phase.
+	if((rand() < (1/10)) && (survival_wave >= (last_masscas_round + 3)))
+		stage = TUTORIAL_HM_PHASE_NIGHTMARE
+		// increases difficulty by 2 levels, but not beyond the max.
+		for(var/i in 1 to 2)
+			var/current_difficulty = survival_difficulty
+			if(current_difficulty != TUTORIAL_HM_INJURY_SEVERITY_MAXIMUM)
+				survival_difficulty = next_in_list(current_difficulty, difficulties)
+		// heightened patient NPC spawn rates, from 4-6
+		min_survival_agents = 4
+		max_survival_agents = 6
+		playsound(tutorial_mob.loc, 'sound/effects/siren.ogg', 50)
+		message_to_player("Warning! Mass-Casualty event detected!")
+	// 50% chance per wave of increasing difficulty by one step
+	// two round grace period from start
+	else if((rand() < TUTORIAL_HM_DIFFICULTY_INCREASE) && !(survival_wave <= 2))
```

Comment:
this basically says; `else if ((rand(0, 1) < (1/2) && !(survival_wave <= 2)`
why would you do it like this.
```suggestion
	else if(prob(50) && !(survival_wave <= 2))
```